Buffs/BuffBase.gd

```diff
@@ -0,0 +1,37 @@
+extends Resource
+class_name BuffBase
+## Base buff resource
+##
+## If you want to make a new buff, create a child of this.
+## If the buff applies some sort of instant effect (eg poison), you can use an
+## event function like on_turn_end.
+## If the buff applies persistent changes to the Entity's stats (eg strength), make changes to
+## the stats in get_modified_stats.
+
+
+@export var infinite_duration: bool = false
+@export var buff_turn_duration: int = 3
+@export var buff_power: float = 1.0
```

Comment:
Tags makes sense if we have lots of Cards/Buffs that modify specific things about damage or buffs, but I feel like those will probably be rare cases where you could just create a new child CardBase or BuffBase. Definitely something to consider though if this system becomes unwieldy

Reply:
I think I see what you're saying. So the suggestion is we may want to have a card that removes all buffs that have a certain tag. And buffs that are tagged like this could expand as the project expands -- which would make it annoying to go back and add the removal functionality to old cards if we didn't have the tagging system. Am I understanding properly?

I think if we end up designing cards that can do such a thing, then the tagging system is a great idea. We could even expand tagging to cards as well. For example, a card that deals damage for all cards in your hand that are tagged "Skill". I think this is something we can address in a future PR if we end up needing it

Reply:
Yes that's it. We do not know how many cards like that there would be in the end, but I have hard time believing there wouldn't be at least some - possibly a lot depending on how many cards we will have overall. This could also help with dealing damage if we will have damage of different types, and deciding on buffs and debuffs that should be taken into consideration and whatnot. 

But yes, this is something that probably goes beyond scope of what can/should be modified on the fly in a PR and should probably be it's own PR later